pkg/vsphere/tasks/waiter.go

```diff
@@ -60,6 +72,23 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	var info *types.TaskInfo
 	var backoffFactor int64 = 1
 
+	// initialize the limiter channel once
+	once.Do(func() {
+		inFlight = make(chan struct{}, maxInFlight)
+	})
+
+	// reserve the spot
+	inFlight <- struct{}{}
```

Comment:
This way will block the task directly without any message out.
Is thinking if do not want to have one more lock for available spots, might just get op before this, and log out message before and after get the spot.
So we could figure out the time gap to query one available task spot, to help debug.